### PR TITLE
[16.0][FIX] cannot re-assign move after applying inventory

### PR DESCRIPTION
**Pull request by nguyenminhchien**
Related issue: https://github.com/OCA/stock-logistics-workflow/issues/1380

### CODE REVIEW

**Comment by rousseldenis** [comment]
@nguyenminhchien LGTM. Could you maybe add a test that explain that change ?

**Comment by nguyenminhchien** [comment]
@rousseldenis : updated.

File: stock_move_free_reservation_reassign/models/stock_quant.py [hunk 1]
```diff
@@ -25,5 +25,7 @@ def _apply_inventory(self):
         ]._get_move_free_reservation_ids() as move_to_reassign_ids:
             res = super()._apply_inventory()
         if move_to_reassign_ids:
-            self.env["stock.move"].browse(move_to_reassign_ids)._action_assign()
+            moves = self.env["stock.move"].browse(move_to_reassign_ids)
+            if not self.env.context.get("assign_in_inventory_mode"):
```
Comment (rousseldenis) [inline stock_move_free_reservation_reassign/models/stock_quant.py]: @nguyenminhchien I don't get why you need a second context parameter.

The ir.config_parameter is sufficent.

Comment (nguyenminhchien) [reply stock_move_free_reservation_reassign/models/stock_quant.py]: it's painful to add the context "assign_in_inventory_mode". Remove inventory_mode (`inventory_mode=False`) is a must to keep this module works, making the current tests green are enough to explain the change.
So, i will revert this change, no need more unittest, what do you think?

**Review by rousseldenis — APPROVED** [review]

**Review by lmignon — APPROVED** [review]

**Pull request merged**